pkg/kcp/provider/gcp/nuke/new.go

```diff
@@ -0,0 +1,43 @@
+package nuke
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+	nuketypes "github.com/kyma-project/cloud-manager/pkg/kcp/nuke/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func New(stateFactory StateFactory) composed.Action {
+	return func(ctx context.Context, st composed.State) (error, context.Context) {
+
+		state, err := stateFactory.NewState(ctx, st.(nuketypes.State))
+		if err != nil {
+			obj := st.Obj().(*v1beta1.Nuke)
+			return composed.UpdateStatus(obj).
+				SetExclusiveConditions(metav1.Condition{
+					Type:    v1beta1.ConditionTypeError,
+					Status:  metav1.ConditionTrue,
+					Reason:  v1beta1.ReasonGcpError,
+					Message: err.Error(),
+				}).
+				SuccessError(composed.StopAndForget).
+				SuccessLogMsg(fmt.Sprintf("Error creating new GCP Nuke state: %s", err)).
+				Run(ctx, st)
+		}
+		return composed.ComposeActions(
+			"gcpNuke",
+			loadNfsBackups,
+			providerResourceStatusDiscovered,
+			deleteNfsBackup,
+			providerResourceStatusDeleting,
+			providerResourceStatusDeleted,
+			checkIfAllProviderResourcesDeleted,
+			// continue to parent action
+			func(ctx context.Context, state composed.State) (error, context.Context) {
+				return nil, nil
```

Comment:
Don't lose the context, return it. Think it's not modified, but it's a good habit to always return it.
```suggestion
				return nil, ctx
```